src/main.cpp

```diff
@@ -64,8 +64,49 @@ static void servoSetUs(unsigned int servo, int usec)
     g_OutputsUs[servo] = usec;
 }
 
+
+static void outputFailsafeValues()
+ {
+    for (unsigned int out=0; out<NUM_OUTPUTS; ++out)
+        {
+            if (OUTPUT_FAILSAFE[out] == fsaNoPulses)
+                servoSetUs(out, 0);
+            else if (OUTPUT_FAILSAFE[out] != fsaHold)
+                servoSetUs(out, OUTPUT_FAILSAFE[out]);
+            // else fsaHold does nothing, keep the same value
+        }
+}
+
+
+#if defined(ARMSWITCH)
+    short armCount = 0;
+#endif
+
 static void packetChannels()
 {
+    #if defined(ARMSWITCH)
+        // If channel 5 doesn't send "arm" signal (value 2000), output failsafe only
+        if (crsf.getChannel(5) != 2000)
+        {
+            outputFailsafeValues();
+            armCount = 0;
+            return;
+        } 
+        else
+        {
+            // Require at least 4 packets with "arm" signal, in order to
+            // prevent accidental arming due to corrupt signals, similar
+            // to Betaflight
+            if (armCount < 4)
+            {
+                outputFailsafeValues();

```

Comment:
Somehow merge these two bits of code that end up arriving at outputFailsafeValues(), do not duplicate.